src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECAndroid.ImportExport.cs

```diff
@@ -0,0 +1,183 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+namespace System.Security.Cryptography
+{
+    internal sealed partial class ECAndroid
+    {
+        public int ImportParameters(ECParameters parameters)
+        {
+            SafeEcKeyHandle key;
+
+            parameters.Validate();
+
+            if (parameters.Curve.IsPrime)
+            {
+                key = ImportPrimeCurveParameters(parameters);
+            }
+            else if (parameters.Curve.IsCharacteristic2)
+            {
+                key = ImportCharacteristic2CurveParameters(parameters);
+            }
+            else if (parameters.Curve.IsNamed)
+            {
+                key = ImportNamedCurveParameters(parameters);
+            }
+            else
+            {
+                throw new PlatformNotSupportedException(
+                    SR.Format(SR.Cryptography_CurveNotSupported, parameters.Curve.CurveType.ToString()));
+            }
+
+            if (key == null || key.IsInvalid)
+            {
+                throw new CryptographicException();
+            }
+
+            FreeKey();
+            _key = new Lazy<SafeEcKeyHandle>(key);
+            return KeySize;
+        }
+
+        public static ECParameters ExportExplicitParameters(SafeEcKeyHandle currentKey, bool includePrivateParameters) =>
+            ExportExplicitCurveParameters(currentKey, includePrivateParameters);
+
+        public static ECParameters ExportParameters(SafeEcKeyHandle currentKey, bool includePrivateParameters)
+        {
+            ECParameters ecparams;
+            if (Interop.AndroidCrypto.EcKeyHasCurveName(currentKey))
```

Comment:
This is going to ask for the name, see it's not null, throw away the string, then ask for it again.

Consider changing this to a flow where you ask for the name, and if it's non-null pass it into ExportNamedCurveParameters, otherwise you call ExportExplicit.